### PR TITLE
[react-native] added _getValue function definition to animated values

### PR DESCRIPTION
**Pull request by AlmirKadric**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -7780,6 +7780,11 @@ export namespace Animated {
          */
         setValue(value: number): void;
 
+		/**
+		 * Gets the current animated value. Usefule when
```
Comment (alloy) [inline types/react-native/index.d.ts]: This comment is incomplete.

**Review by alloy — CHANGES_REQUESTED** [review]
Being prepended with an underscore, this looks to be a private API, is that correct? If that’s the case, I don’t think we should advertise it as if it were public.

In any case, please fill out a URL to the documentation or source, as asked by the PR template.

File: types/react-native/index.d.ts (hunk 1 above)
Comment (AlmirKadric) [reply types/react-native/index.d.ts]: woops
must have ctrl+z'd accidentally

**Comment by AlmirKadric** [comment]
@alloy yes it appears to be a private API however and there is no documentation for the API from what I can see. However if you check the react-native eco-system it is the only way to get the animated value and is used by a number of modules, including react-native internal modules.

> In any case, please fill out a URL to the documentation or source, as asked by the PR template.
As there is no official documentation for this specific API not sure what I should be linking to.

**Comment by alloy** [comment]
I’m really torn about this.

@huhuanming @iRoachie @timwangdev @kamal @nelyousfi what are all your thoughts?

**Comment by iRoachie** [comment]
We really shouldn't expose methods not declared in the public api. Private apis change all the time between releases and there's no way we'd know when a private api is updated.

However he's right in saying the solutions provided online make use of  `this._getValue()`.

**Comment by AlmirKadric** [comment]
@alloy @iRoachie I took a deeper look and found the following:
 - https://github.com/facebook/react-native/issues/3633
 - https://facebook.github.io/react-native/docs/animations.html#responding-to-the-current-animation-value

There are issues with getting the current value of an AnimatedValue due to native side optimizations, however there are times when using the value is the only way to get something working. This is why it is a private API rather than a public one. I don't see that API in particular going away anytime soon.

**Comment by iRoachie** [comment]
Yep @AlmirKadric these are the two articles I came across as well this [stack overflow answer](https://stackoverflow.com/questions/41932345/get-current-value-of-animated-value-react-native). According to the docs, it seems like although not recommended; they've provided this hatch for special cases

**Comment by timwangdev** [comment]
I agree with @alloy. People should fix their own types if they are using a private api.

**Comment by AlmirKadric** [comment]
@alloy @timwangdev how would you suggest doing that? we can't patch types onto classes. So the type errors will always be errors meaning if you have CI tied to your build process, it will always fail at the compile phase

**Comment by timwangdev** [comment]
@AlmirKadric I believe this one will do the work.
```typescript
type PatchedValue = Animated.Value & { __getValue?: () => number };
const foo: PatchedValue = new Animated.Value(0);
foo.__getValue && foo.__getValue();
```
There is another guide from TypeScript https://www.typescriptlang.org/docs/handbook/declaration-merging.html but I'm not familar with.

**Comment by alloy** [comment]
Yup, @timwangdev’s suggestion is a good one.

Ok, I’m going to close this PR. Thanks for taking the time to make the patch and argue in favour of it, though! 🙏

**Comment by AlmirKadric** [comment]
@timwangdev @alloy i already tried type merging (many hours wasted on trying to get it to work), but it unfortunately doesn't work for adding properties to a class. Classes are the least supported when it comes to type merging and proves to be very challenging. If you could provide me with sample code on how that will work im all ears, but it wont to my knowledge.

**Comment by alloy** [comment]
I don’t think we swept anything under the proverbial rug, we had a discussion and came to a conclusion.

As for a working suggestion, [this](http://www.typescriptlang.org/play/index.html#src=class%20Foo%20%7B%0D%0A%7D%0D%0A%0D%0Atype%20FooWithPrivate%20%3D%20Foo%20%26%20%7B%20_getValue%3A%20()%20%3D%3E%20string%20%7D%20%0D%0A%0D%0Aconst%20x%20%3D%20new%20Foo()%20as%20FooWithPrivate%0D%0Ax._getValue()) seems to work fine.

```ts
class Foo {
}

type FooWithPrivate = Foo & { _getValue: () => string } 

const x = new Foo() as FooWithPrivate
x._getValue()
```

**Comment by AlmirKadric** [comment]
@alloy thanks for the follow up, apologies for my poor choice of words i didn't mean it in an aggressive way. I was simply asking you to spend some time to suggest some alternatives as to how the type could be merged (which you have done) before a decision was made, it was a very one sided decision IMHO.

Moving forward, I do have some reservations with your proposal. Its not quite a global merge onto the existing type, instead its a new type definition with a forced cast to hack on the new property. This is not all that nice and I would dare say an ugly trade off just to avoid defining private properties. Is there no better way?

**Pull request closed without merging**